### ISSUE TITLE
[kothic] add metatypes like `all`, `nodeway` etc to simplify drules

### ISSUE THREAD

**Issue by endim8** [organicmaps/organicmaps#4137]
It would be nice to have metatypes that generate styles for real types, this would massively simplify the styles.mapcss files.

eg.

Going from:
```
node|z13[railway=station][transport=subway][city=lille] {icon-image: subway-lille-s.svg;icon-min-distance: 1;}
node|z14[railway=station][transport=subway][city=lille] {icon-image: subway-lille-s.svg;}
node|z15-16[railway=station][transport=subway][city=lille] {icon-image: subway-lille-m.svg;}
node|z17-[railway=station][transport=subway][city=lille] {icon-image: subway-lille-m.svg;text-optional: true;}

area|z13[railway=station][transport=subway][city=lille] {icon-image: subway-lille-s.svg;icon-min-distance: 1;}
area|z14[railway=station][transport=subway][city=lille] {icon-image: subway-lille-s.svg;}
area|z15-16[railway=station][transport=subway][city=lille] {icon-image: subway-lille-m.svg;}
area|z17-[railway=station][transport=subway][city=lille] {icon-image: subway-lille-m.svg;text-optional: true;}

```

to 

```
nodearea|z13[railway=station][transport=subway][city=lille] {icon-image: subway-lille-s.svg;icon-min-distance: 1;}
nodearea|z14[railway=station][transport=subway][city=lille] {icon-image: subway-lille-s.svg;}
nodearea|z15-16[railway=station][transport=subway][city=lille] {icon-image: subway-lille-m.svg;}
nodearea|z17-[railway=station][transport=subway][city=lille] {icon-image: subway-lille-m.svg;text-optional: true;}
```

**Comment by biodranik** [comment]
That is a good idea. Take a look at https://github.com/organicmaps/kothic, I think this can be done easy.

**Comment by pastk** [comment]
`all` probably works already, check wildcard usage in `defaults_new.mapcss`, e.g.
```
*::int_name
{
  text-offset: 1;
}

*[amenity],
*[office],
...
{
  text-position: center;
}
```

But in your subway sample it shouldn't be `*`, but node + area w/o line.

**Comment by pastk** [comment]
Actually I think we can drop `node/line/area` type definitions in mapcss completely as they're redundant.
E.g. there are no `node/line/area` types in compiled drules files.
E.g. if there is an area-specific attribute defined like `fill-color:` then kothic will create a corresponding `area` element in drules regardless of the type defined.

Right now its possible to e.g. delete all duplicate `node` definitions and leave `area` ones only for some types - and nothing will change. However in some cases its more tricky as e.g. priorities are different for node and area elements.

CC @vng

**Comment by vng** [comment]
Yes, I know that. It was a long-long time ago, I can't say for sure why did we make like that ..

Probably, better to keep node|line|area, but also add something like "poi" selector to avoid node and area copy-paste.
Don't you think that it will be useful in future (like possible different icons for node or area POIs)?

BTW, does comma works? Like: node,area|z..|[amenity]

**Comment by pastk** [comment]
> 
> Probably, better to keep node|line|area, but also add something like "poi" selector to avoid node and area copy-paste. Don't you think that it will be useful in future (like possible different icons for node or area POIs)?

ATM defining different icons doesn't work, even "no area icons" doesn't work, see #4466

**Comment by pastk** [comment]
> Probably, better to keep node|line|area, but also add something like "poi" selector to avoid node and area copy-paste. Don't you think that it will be useful in future (like possible different icons for node or area POIs)?

Do you have a plausible use case in mind?
If not, then its better to remove those attributes - they haven't been used for years, but were polluting the styles.

> 
> BTW, does comma work? Like: node,area|z..|[amenity]

Nope, currently only wildcards are supported, e.g.
- without zoom:
```
[natural=wetland],
{z-index: 43;}
```
(or `*[natural=wetland]`, but its less readable)

- with zoom:
```
*|z11-[natural=wetland],
{fill-color: @wetland;}
```
